### PR TITLE
feat: download .wpress backups from cloud storage URLs

### PR DESCRIPTION
**Pull request by dugyen**
## Summary

- **CloudDownloader** (`src/clouddownloader.{h,cpp}`): downloads .wpress files from remote HTTPS URLs to a temp file via `QNetworkAccessManager`. Normalizes share links from Google Drive, Dropbox, OneDrive, Box, and pCloud into direct download URLs. Mega (mega.nz) uses a two-step flow: parse the AES-128-CTR key from the URL fragment (never sent to any server), call the Mega public API for a CDN URL, then decrypt the encrypted stream on-the-fly with OpenSSL `EVP_aes_128_ctr()`.
- **Drop overlay** (`src/dropoverlay.{h,cpp}`): accepts remote URLs in addition to local .wpress files. Two-path drag handling — `text/uri-list` primary, `text/plain` fallback — so cloud web-UI drags work and Mega `#KEY` fragments are preserved (`text/uri-list` strips fragments per RFC 2483).
- **URL dialog** (`src/urlopendialog.{h,cpp}`): File → Open from URL… (⌘⇧O / Ctrl+Shift+O) lets the user paste any cloud share link or direct URL.
- **MainWindow wiring**: `urlDropped` → `openBackupFromUrl`, download progress bar with indeterminate mode when Content-Length is unknown, re-entrancy guard, temp file cleanup on clear/re-download.
- **24 new tests** in `tst_clouddownloader.cpp` covering `isRemoteUrl`, `normalizeUrl` (per provider), and `parseMegaUrl` (key derivation, legacy format, edge cases). All 121 tests pass.

### Supported providers

| Provider | Normalization |
|---|---|
| Google Drive | `/file/d/{ID}/view` and `/open?id={ID}` → `/uc?export=download&id={ID}&confirm=t` |
| Dropbox | `dl=0` → `dl=1` |
| OneDrive / 1drv.ms | Appends `download=1` |
| Box | `/s/{code}` → `/shared/static/{code}` |
| pCloud | `publink/show?code=X` → `publink/code?code=X&forcedownload=1` |
| Mega | Two-step: API call for CDN URL + AES-128-CTR decryption (key from URL fragment) |
| S3 / GCS / Azure / DO Spaces | Pass-through (pre-signed or public URLs work as-is) |

### Security notes

- The Mega decryption key lives in the URL fragment and is **never** transmitted to any server.
- No telemetry, no background connections. `QNetworkAccessManager` is only used when the user explicitly provides a URL.
- Downloaded files land in `QTemporaryFile` in the system temp dir and are cleaned up on clear or re-download.
- FTP is explicitly rejected — Qt 6 removed FTP from `QNetworkAccessManager`.

## Test plan

- [ ] Drag a local `.wpress` file — still works as before
- [ ] Drag a Google Drive share link from Chrome → overlay accepts, file downloads
- [ ] Drag a Dropbox share link → downloads with `dl=1`
- [ ] Drag a Mega link → two-step download + AES decryption produces valid `.wpress`
- [ ] File → Open from URL → paste a OneDrive link → downloads
- [ ] Paste an invalid URL → OK button stays disabled
- [ ] Cancel mid-download → temp file is cleaned up
- [ ] Drop a second URL while downloading → ignored (re-entrancy guard)
- [ ] `ctest --output-on-failure` — all 121 tests pass (97 existing + 24 new)
- [ ] `clang-format --dry-run --Werror` — no violations

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -343,6 +365,73 @@ void MainWindow::openBackupFile(const QString &filename)
...
+        return;
+    openBackupFromUrl(dlg.url());
+}
+
+void MainWindow::setDownloadingState(bool downloading)
+{
+    m_downloading = downloading;
+    ui->openBackupButton->setEnabled(!downloading);
+    ui->openUrlButton->setEnabled(!downloading);
+    ui->extractBackupButton->setEnabled(!downloading);
```
[the remark this answers is not in the document]
Comment (dugyen) [reply src/mainwindow.cpp]: Fixed in 4a78aa0 — `setDownloadingState()` now gates the extract button on `!backupFilename.isEmpty()` (matching the clearButton line beside it), so a failed download from the empty state no longer leaves **Extract** enabled with no backup loaded.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/clouddownloader.cpp [hunk 2]
```diff
@@ -0,0 +1,553 @@
...
+    cleanupTempFile();
+}
+
+// ── URL helpers ────────────────────────────────────────────────────────────────
+
+bool CloudDownloader::isRemoteUrl(const QUrl &url)
+{
+    // Qt 6 removed FTP support from QNetworkAccessManager; only HTTP(S) accepted.
+    const QString scheme = url.scheme().toLower();
+    return scheme == QLatin1String("http") || scheme == QLatin1String("https");
```
[the remark this answers is not in the document]
Comment (dugyen) [reply src/clouddownloader.cpp]: Fixed in 4a78aa0 — `isRemoteUrl()` now accepts `https://` only; plain `http://` is rejected and the unit test was flipped accordingly.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/clouddownloader.cpp [hunk 3]
```diff
@@ -0,0 +1,553 @@
...
+        QUrl dl = url;
+        dl.setQuery(q);
+        return dl;
+    }
+
+    // pCloud is handled inside download() (needs API call to resolve CDN URL).
+    // Amazon S3, DigitalOcean Spaces, Google Cloud Storage, Azure Blob —
+    // direct / pre-signed HTTPS URLs work as-is.
+    // Mega is handled inside download(), not here (needs API call + decryption).
+    return url;
```
[the remark this answers is not in the document]
Comment (dugyen) [reply src/clouddownloader.cpp:141]: No change needed — OneDrive and Box support was removed earlier in this branch, so there's no code path for them to fall through. (OneDrive personal shares are SharePoint-migrated and need a JS guest session; Box's `/shared/static` endpoint now 404s — both verified against live links.) Current providers: Google Drive, Dropbox, pCloud, Mega, WeTransfer.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: src/clouddownloader.cpp [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+// place. For non-Mega downloads the bytes are written through unchanged.
+void CloudDownloader::writeChunk(QByteArray data)
+{
+    if (!m_tempFile)
+        return;
+
+    if (m_isMega && m_aesCtr)
+        mega_aes_ctr_xcrypt(m_aesCtr, reinterpret_cast<unsigned char *>(data.data()), static_cast<size_t>(data.size()));
+
+    m_tempFile->write(data);
```
[the remark this answers is not in the document]
Comment (dugyen) [reply src/clouddownloader.cpp]: Fixed in f030433 — `writeChunk()` now checks `QTemporaryFile::write()` and records a short write; `onFinished()` fails with a disk-write error and removes the partial file instead of opening a truncated `.wpress`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
